shared/common-adapters/avatar.native.js

```diff
@@ -10,31 +10,36 @@ import {iconMeta} from './icon.constants'
 
 type State = {
   avatarLoaded: boolean,
+  errored: boolean,
 }
 
 class Avatar extends Component<void, Props, State> {
   state: State;
 
   constructor (props: Props) {
     super(props)
-    this.state = {avatarLoaded: false}
+    this.state = {avatarLoaded: false, errored: false}
   }
 
   componentWillReceiveProps (nextProps: Props) {
     const url = shared.createAvatarUrl(this.props)
     const nextUrl = shared.createAvatarUrl(nextProps)
 
     if (url !== nextUrl) {
-      this.setState({avatarLoaded: false})
+      this.setState({avatarLoaded: false, errored: false})
     }
   }
 
   render () {
     const {size} = this.props
-    const uri = {uri: shared.createAvatarUrl(this.props)}
+    const uri = shared.createAvatarUrl(this.props)
     const propsOpacity = this.props.hasOwnProperty('opacity') ? this.props.opacity : 1.0
     const opacity = this.state.avatarLoaded ? propsOpacity : 0
 
+    const showNoAvatar = (!this.props.loadingColor && (!this.state.avatarLoaded || !uri)) ||
```

Comment:
This boolean expression is a bit hard to parse. Can it be simplified, or broken down into named subexpressions?